frontend/src/components/authentication/CreateAccountForm.jsx

```diff
@@ -0,0 +1,328 @@
+import React, { Component } from "react";
+import LOCALIZE from "../../text_resources";
+import validateName, { validateEmail, validatePassword } from "../../helpers/regexValidator";
+import "../../css/create-account-form.css";
+
+const styles = {
+  createAccountContent: {
+    padding: "12px 32px 0 32px"
+  },
+  inputTitle: {
+    padding: "12px 0 6px 0",
+    fontWeight: "bold"
+  },
+  inputs: {
+    width: "100%",
+    padding: "3px 6px 3px 6px",
+    borderRadius: 4
+  },
+  inputForNames: {
+    width: 190,
+    padding: "3px 6px 3px 6px",
+    borderRadius: 4
+  },
+  iconForNames: {
+    color: "#278400",
+    position: "absolute",
+    margin: "8px 0 0 170px"
+  },
+  iconForOtherFields: {
+    color: "#278400",
+    position: "absolute",
+    margin: "8px 0 0 370px"
+  },
+  loginBtn: {
+    width: 150,
+    display: "block",
+    margin: "24px auto"
+  },
+  validationError: {
+    color: "red",
+    marginTop: 6
+  }
+};
+
+let SUBMIT_BTN_DISABLED;
+
+class CreateAccountForm extends Component {
+  state = {
+    isFirstPasswordLoad: true,
+    isValidFirstName: false,
+    firstNameClassStyle: "valid-field",
+    isValidLastName: false,
+    lastNameClassStyle: "valid-field",
+    isValidEmail: false,
+    emailClassStyle: "valid-field",
+    isValidPassword: false,
+    passwordClassStyle: "valid-field",
+    isValidPasswordConfirmation: false,
+    passwordConfirmationClassStyle: "valid-field"
+  };
+
+  firstNameValidation = () => {
+    const firstName = document.getElementById("first-name").value;
+    const isValide = validateName(firstName);
+    if (isValide) {
+      this.setState({ isValidFirstName: true, firstNameClassStyle: "valid-field" });
+    } else {
+      this.setState({ isValidFirstName: false, firstNameClassStyle: "invalid-field" });
+    }
+  };
+
+  lastNameValidation = () => {
+    const lastName = document.getElementById("last-name").value;
+    const isValide = validateName(lastName);
+    if (isValide) {
```

Comment:
By not holding classes in state, these functions become much simpiler:
`this.setState({isValidLastName: validateName(lastname)});`

Reply:
Fixed in commit #https://github.com/code-for-canada/project-thundercat/pull/156/commits/d0a96d54d1e34be849fa1e01798e2cc39ef92d02